src/cudamatrix/cu-device.h

```diff
@@ -26,6 +26,7 @@
 #if HAVE_CUDA == 1
 
 #include <cublas_v2.h>
+#include <cudnn.h>
```

Comment:
If the installation is easy (e.g. if they let us distribute it), we might
just make the decision that whenever we support CUDA, we also require CUDNN
support.  That would reduce the number of separate cases of compilation
that have to be tested.  At this point I think we haven't made the decision
to go with CUDNN at all, this is somewhat experimental.

Dan

On Mon, Mar 28, 2016 at 3:17 PM, Daniel Galvez <notifications@github.com>
wrote:

> In src/cudamatrix/cu-device.h
> <https://github.com/kaldi-asr/kaldi/pull/604#discussion_r57617709>:
>
> > @@ -26,6 +26,7 @@
> >  #if HAVE_CUDA == 1
> >
> >  #include <cublas_v2.h>
> > +#include <cudnn.h>
>
> You should wrap this, and I think all of the code that uses the cudnn
> context in cu-device.cc in this:
>
> #ifdef HAVE_CUDNN
> code goes here...
> #endif
>
> You could define the HAVE_CUDNN macro somewhere in src/configure by adding
> a line like "CXX_FLAGS += -DHAVE_CUDNN" to kaldi.mk
>
> The point is that I don't think we can reasonably assume most users of
> Kaldi will want or need cudnn. So we have to have these ugly macros to
> support compiling both with and without cudnn.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/604/files/b1e82b2faceed80758f8fcb33e8ba203057d8252..722c7e5a927305e2e02ba6735a2c14761fed6d05#r57617709>
>